### PR TITLE
Readonly settings columns in views

### PR DESCRIPTION
**Pull request by adrinr**
## Description
CRUD view updates to allow persisting `readonly` configuration for columns. Updating this config will be a paid feature.

Pro PR:
- https://github.com/Budibase/budibase-pro/pull/298

## Addresses
- [BUDI-8284 - [Readonly views] Backend support for new column configuration](https://linear.app/budibase/issue/BUDI-8284/[readonly-views]-backend-support-for-new-column-configuration)

### CODE REVIEW

File: packages/server/src/api/routes/tests/viewV2.spec.ts [hunk 1]
```diff
@@ -450,6 +676,67 @@ describe.each([
...
+        schema: {
+          Price: {
+            visible: true,
+            readonly: true,
+          },
+        },
+      })
+
+      mocks.licenses.useCloudFree()
+      await config.api.viewV2.create(view, {
```
Comment (samwho) [inline packages/server/src/api/routes/tests/viewV2.spec.ts:694]: Is `create` correct here?

File: packages/server/src/sdk/app/views/index.ts [hunk 2]
```diff
@@ -31,14 +35,62 @@ export async function getEnriched(viewId: string): Promise<ViewV2Enriched> {
...
 
+async function guardViewSchema(
+  tableId: string,
+  viewSchema?: Record<string, ViewUIFieldMetadata>
+) {
+  if (!viewSchema || !Object.keys(viewSchema).length) {
+    return
+  }
+  const table = await sdk.tables.getTable(tableId)
+  if (viewSchema) {
```
Comment (samwho) [inline packages/server/src/sdk/app/views/index.ts:46]: You don't need this condition, you check for `!viewSchema` above and return early.

File: packages/server/src/api/routes/tests/viewV2.spec.ts (hunk 1 above)
Comment (adrinr) [reply packages/server/src/api/routes/tests/viewV2.spec.ts:694]: No, good catch!

**Review by samwho — APPROVED** [review]
Absolutely fantastic tests. 👏

**Pull request merged**